data/transactions/logic/fields_test.go

```diff
@@ -17,12 +17,177 @@
 package logic
 
 import (
+	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/test/partitiontest"
 )
 
 func TestArrayFields(t *testing.T) {
 	require.Equal(t, len(TxnaFieldNames), len(TxnaFieldTypes))
 	require.Equal(t, len(txnaFieldSpecByField), len(TxnaFieldTypes))
 }
+
+// ensure v2+ fields fail in TEAL assembler and evaluator on a version before they introduced
+// ensure v2+ fields error in v1 program
+func TestGlobalFieldsVersions(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	var fields []globalFieldSpec
+	for _, fs := range globalFieldSpecs {
+		if fs.version > 1 {
+			fields = append(fields, fs)
+		}
+	}
+	require.Greater(t, len(fields), 1)
+
+	consensusParamsByLogicVersion := make(map[uint64]config.ConsensusParams, LogicVersion)
+	for v := uint64(0); v <= LogicVersion; v++ {
+		consensusParamsByLogicVersion[v] = defaultEvalProtoWithVersion(v)
+	}
```

Comment:
No reason for the map anymore, right?  Just invoke defaultEvalProtoWIthVersion where you would do the lookup and avoid dealing with require.True(t, ok) afterward.

Reply:
true, fixed